hackathon_site/event/views.py

```diff
@@ -226,3 +227,34 @@ class TeamListView(mixins.ListModelMixin, generics.GenericAPIView):
 
     def get(self, request, *args, **kwargs):
         return self.list(request, *args, **kwargs)
+
+
+class ProfileDetailView(mixins.RetrieveModelMixin, generics.GenericAPIView):
+
+    queryset = Profile.objects.all()
+    serializer_class = ProfileModifySerializer
+    lookup_field = "id"
+    permission_classes = [FullDjangoModelPermissions | UserHasProfile]
+
+    @transaction.atomic
+    def patch(self, request, *args, **kwargs):
+        if self.request.user.has_perm("event.change_profile"):
+            profile = self.get_object()
+        elif request.user.profile:
+            profile = request.user.profile
+        else:
+            return HttpResponseForbidden()
+
+        data = request.data
+        profile.id_provided = data["id_provided"]
```

Comment:
these two fields are something that only admins should be able to check (id provided when they visit the tech team, attended when they sign in on the day-of).

Do you think it would be easier if we leave this endpoint as only accessible to admins, and make a separate `PATCH /event/profiles/profile/`  that regular users can use? Similar to we have `/event/users/user/` and `/events/teams/team/` which deal with the currently logged in user and their team.

Reply:
yeah I think that's a lot cleaner